samples/drivers/led_ws2812/README.rst

```diff
@@ -0,0 +1,129 @@
+.. _led_ws2812_sample:
+
+WS2812 Sample Application
+#########################
+
+Overview
+********
+
+This sample application demonstrates basic usage of the WS2812 LED
+strip driver, for controlling LED strips using WS2812, WS2812b,
+SK6812, and compatible driver chips.
+
+Requirements
+************
+
+.. _NeoPixel Ring 12 from AdaFruit: https://www.adafruit.com/product/1643
+.. _74AHCT125: https://cdn-shop.adafruit.com/datasheets/74AHC125.pdf
+
+- LED strip using WS2812 or compatible, such as the `NeoPixel Ring 12
+  from AdaFruit`_.
+
+- Zephyr board with SPI master driver. SPI communications must use 5V
+  signalling, which may require a level translator, such as the
+  `74AHCT125`_.
+
+- 5V power supply.
+
+Wiring
+******
+
+#. Ensure your Zephyr board, the 5V power supply, and the LED strip
+   share a common ground.
+#. Connect the MOSI pin of your board's SPI master to the data input
+   pin of the first WS2812 IC in the strip.
+#. Connect the 5V power supply pin to the 5V input of the LED strip.
+
+Building and Running
+********************
+
+.. _blog post on WS2812 timing: https://wp.josh.com/2014/05/13/ws2812-neopixels-are-not-so-finicky-once-you-get-to-know-them/
+
+The sample application is located at ``samples/drivers/led_ws2812/``
+in the Zephyr source tree.
+
+Configure For Your LED Strip
+----------------------------
+
+The first thing you need to do is make sure that the driver is
+configured to match the particular LED chips you're using. For
+example, some chips have a "green, red, blue" color ordering on their
+data pin, while others have "red, green, blue, white". Check your
+chip's datasheet for your configuration (though given the number of
+competing implementations and knock-offs, some trial and error may be
+necessary).
+
+To make sure the driver is set up properly for your chips, check the
+values of the following configuration options:
+
+- ``CONFIG_WS2812_RED_ORDER``
```

Comment:
You can use the notation ``:option:`CONFIG_WS8212_RED_ORDER` ``to create a link to the configuration option's documentation for all of these.